🐛 Re-implementing truncation of long links in Recent Links

Sean-Stilwell opened the pull request:
# Pull Request

## Description
<!-- A brief description of what this PR does -->

Recent links were intended to truncate to be the same size as the other navigation boxes, but this was not functional.

This PR caps the length of the labels at 40 characters.

### Before

![image](https://github.com/user-attachments/assets/8b6018d6-39e8-4a14-8e33-4a13c9734d81)

### After

![image](https://github.com/user-attachments/assets/38ec8d54-3ff0-4304-9c67-955631f0c3e6)

## Related Issues
<!-- List any related issues or tickets -->

TASK 8502

## Changes
<!-- List the key changes in this PR -->

- Replaces current recent activity display to use a DHButton
- Adds truncation using styling on the DHButton

## Testing
<!-- Describe the tests that were run or any QA steps that were taken -->

- Locally tested

## Checklist
<!-- Ensure the following tasks are complete -->

- [x] Code follows dotnet coding standards
- [x] Tests added/updated to cover changes

Sean-Stilwell committed:
Making truncation of long recent links functional

ErikApption commented:
Are we sure we are not breaking any accessibility here? We are not using a style to truncate but effectively truncating the names. I would look at using css

Sean-Stilwell committed:
Switching recent activity to use buttons with truncation in styling, removing separate icon objects

Sean-Stilwell committed:
Reverting GetLabel

Sean-Stilwell committed:
Merge branch 'develop' into ss/truncate-recent-links

Sean-Stilwell commented:
> Are we sure we are not breaking any accessibility here? We are not using a style to truncate but effectively truncating the names. I would look at using css

Just to be safe, I've changed the links to be our accessible `DHButton` objects and moved the icon to be `StartIcon`. Truncation is now done using styling too.

![image](https://github.com/user-attachments/assets/f83961c4-862d-439d-ba1e-b1203fbf897a)

ErikApption committed:
Merge branch 'develop' into ss/truncate-recent-links

Sean-Stilwell committed:
Adding tooltip

Sean-Stilwell committed:
Merge branch 'ss/truncate-recent-links' of https://github.com/ssc-sp/datahub-portal into ss/truncate-recent-links

Sean-Stilwell committed:
Switching to boring tooltip
